### PR TITLE
[Merged by Bors] - feat: port RingTheory.Multiplicity

### PR DESCRIPTION
**Pull request by xroblot**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by xroblot** [comment]
@mattrobball Looks like we started to work on the same file at the same time. I hope I didn't messed up with your PR. 

In any case, this file is all yours now ;)

**Comment by mattrobball** [comment]
@xroblot Sorry! I have to step away for a bit. I pushed where I got to. Feel free to work on it please.

File: Mathlib/RingTheory/Multiplicity.lean [hunk 1]
```diff
@@ -0,0 +1,672 @@
...
+
+protected theorem mul' {p a b : α} (hp : Prime p) (h : (multiplicity p (a * b)).Dom) :
+    get (multiplicity p (a * b)) h =
+      get (multiplicity p a) ((finite_mul_iff hp).1 h).1 +
+        get (multiplicity p b) ((finite_mul_iff hp).1 h).2 := by
+  have hdiva : p ^ get (multiplicity p a) ((finite_mul_iff hp).1 h).1 ∣ a := pow_multiplicity_dvd _
+  have hdivb : p ^ get (multiplicity p b) ((finite_mul_iff hp).1 h).2 ∣ b := pow_multiplicity_dvd _
+  have hpoweq :
+    p ^
+        (get (multiplicity p a) ((finite_mul_iff hp).1 h).1 +
```
Comment (RemyDegenne) [inline Mathlib/RingTheory/Multiplicity.lean:558]:
```suggestion
    p ^ (get (multiplicity p a) ((finite_mul_iff hp).1 h).1 +
```

File: Mathlib/RingTheory/Multiplicity.lean [hunk 2]
```diff
@@ -0,0 +1,672 @@
...
+          get (multiplicity p b) ((finite_mul_iff hp).1 h).2) =
+      p ^ get (multiplicity p a) ((finite_mul_iff hp).1 h).1 *
+        p ^ get (multiplicity p b) ((finite_mul_iff hp).1 h).2 :=
+    by simp [pow_add]
+  have hdiv :
+    p ^
+        (get (multiplicity p a) ((finite_mul_iff hp).1 h).1 +
```
Comment (RemyDegenne) [inline Mathlib/RingTheory/Multiplicity.lean:565]:
```suggestion
    p ^ (get (multiplicity p a) ((finite_mul_iff hp).1 h).1 +
```

File: Mathlib/RingTheory/Multiplicity.lean [hunk 3]
```diff
@@ -0,0 +1,672 @@
...
+          get (multiplicity p b) ((finite_mul_iff hp).1 h).2) ∣
+      a * b :=
+    by rw [hpoweq]; apply mul_dvd_mul <;> assumption
+  have hsucc :
+    ¬p ^
+          (get (multiplicity p a) ((finite_mul_iff hp).1 h).1 +
```
Comment (RemyDegenne) [inline Mathlib/RingTheory/Multiplicity.lean:571]:
```suggestion
    ¬p ^ (get (multiplicity p a) ((finite_mul_iff hp).1 h).1 +
```

**Comment by RemyDegenne** [comment]
Thanks!
bors r+

**Pull request closed without merging**